distributed/core.py

```diff
@@ -221,15 +221,22 @@ def start_pcs():
     def stop(self):
         if not self.__stopped:
             self.__stopped = True
-            if self.listener is not None:
+            for listener in self.listeners:
                 # Delay closing the server socket until the next IO loop tick.
                 # Otherwise race conditions can appear if an event handler
                 # for an accept() call is already scheduled by the IO loop,
                 # raising EBADF.
                 # The demonstrator for this is Worker.terminate(), which
                 # closes the server socket in response to an incoming message.
                 # See https://github.com/tornadoweb/tornado/issues/2069
-                self.io_loop.add_callback(self.listener.stop)
+                self.io_loop.add_callback(listener.stop)
+
+    @property
+    def listener(self):
+        if self.listeners:
+            return self.listeners[0]
+        else:
+            return None
```

Comment:
Should this raise some kind of warning as it will limit folks to only accessing one listener in the case that they have multiple listeners.

Reply:
I think that this is likely ok?  In principle we should stop using `listener` any longer, and should always iterate over listeners.  This is here mostly for backwards compatibility and convenience.  Given that this is likely to be ok 99% of the time I'm inclined not to raise a warning.